codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -224,6 +224,56 @@ if [[ -n $S2N_ENSURE_WITH_INVALID_ERROR_CODE ]]; then
   printf "$S2N_ENSURE_WITH_INVALID_ERROR_CODE\n\n"
 fi
 
+#############################################
+# Assert tests don't specify the "default" security policy.
+#
+# Since the "default" policies are subject to change, tests should instead specify
+# an immutable numbered policy to avoid unwanted testing behavior.
+#############################################
+S2N_DEFAULT_SECURITY_POLICY_USAGE=$(find "$PWD" -type f -name "s2n*.c" -not -path "*/bindings/*" -not -path "*/bin/*")
+declare -A KNOWN_DEFAULT_USAGE
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_config.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tls/s2n_security_policies.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_security_policies_test.c"]=5
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_client_hello_test.c"]=2
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_connection_preferences_test.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_config_test.c"]=1
+# "default" string not used for specifying security policy
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_build_test.c"]=1
+KNOWN_DEFAULT_USAGE["$PWD/tests/unit/s2n_client_key_share_extension_test.c"]=2
+
+for file in $S2N_DEFAULT_SECURITY_POLICY_USAGE; do
+  RESULT_NUM_LINES=`grep -n '"default"' $file | wc -l`
+
+  # set default KNOWN_DEFAULT_USAGE value
+  [ -z "${KNOWN_DEFAULT_USAGE["$file"]}" ] && KNOWN_DEFAULT_USAGE["$file"]="0"
+
+  # check if "default" usage is 0 or a known value
+  if [ "${RESULT_NUM_LINES}" != "${KNOWN_DEFAULT_USAGE["$file"]}" ]; then
+    echo "Expected: ${KNOWN_DEFAULT_USAGE["$file"]} Found: ${RESULT_NUM_LINES} usage of '"default"' in $file"
```

Comment:
Some of the other simple mistakes errors offer an explanation when erroring, so that developers know how to work around the issue: https://github.com/aws/s2n-tls/blob/341a69ed6fc4fa8f51c09b9cc477223cec4d8e60/codebuild/bin/grep_simple_mistakes.sh#L100

It might be helpful to say something about using the numbered version of the default policy rather than the default, unless the default policy itself is being tested.